app/src/main/java/com/example/capstoneproject/ui/main/MainFragment.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.capstoneproject.ui.main
+
+import android.os.Bundle
+import android.util.Log
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.capstoneproject.databinding.MainFragmentBinding
+
+// view holders' height should take 1/7 of the screen
+private const val VIEW_HOLDER_SCREEN_PROPORTION = 1.0 / 7.0
+
+class MainFragment : Fragment() {
+
+    companion object {
+        fun newInstance() = MainFragment()
+    }
+
+    private lateinit var binding : MainFragmentBinding
```

Comment:
You may use a nullable binding instead in order to avoid memory leaks
https://developer.android.com/topic/libraries/view-binding#fragments :D